avocado/plugins/assets.py

```diff
@@ -135,6 +135,9 @@ def visit_ClassDef(self, node):  # pylint: disable=C0103
             if self.klass and node.name != self.klass:
                 return
 
+            # if we are parsing a new class, reset the curret method pointer
+            if self.current_klass != node.name:
```

Comment:
Doesn't it make more sense to always reset the `current_method` when setting `current_klass`? My reasoning is that the `ClassDef` will always be visited before the `FunctionDef`, no?

Reply:
Works too, with less caution. I have removed the `if` and force pushed.

Reply:
I double-checked just to make sure, and as far as I could test, `ClassDef` executes before `FunctionDef` and only once for each class.